### ISSUE TITLE
Apple Event timed out when doing range.api.sort()

### ISSUE THREAD

**Issue by tsusoft** [xlwings/xlwings#1955]
#### OS (e.g. Windows 10 or macOS Sierra)
MacOS Monterey v12.4
#### Versions of xlwings, Excel and Python (e.g. 0.11.8, Office 365, Python 3.7)
Python 3.9.5
xlwings 0.27.10
Microsoft Excel for Mac v16.62
#### Describe your issue (incl. Traceback!)
Hi, 
I was working on a sheet with 17000 data rows and 250 columns. I tried to sort the data range by using range.api.sort(), and I encountered Apple Event timed out exception as below.

I know there are closed issues about apple event timed out. I was not able to set ‘timeout’ attribute in sort(), and I tried range.options(chunksize=5000).api.sort(…), but it didn’t work. 

Here are my tries:
1) First, I copied from a raw source, then sort(), it failed. The data remained a half-finished (not sure if it completed or not in exception). I run sort() again based on it, and it passed.  
2) Then, I copied from  raw source each time and run sort(), it failed all the time

I guess it might cost few in the first situation, thus it could pass based on almost-sorted data?

The same issue when I was going to copy/paste large amount of cells. It didn’t work with range().options(chunksize=5000). As workaround, I set Excel visible and real-time-refresh, so that Excel could respond to MacOS. It worked. For even more columns doing copy/paste, I changed codes to run in batch, and it worked. 

But I have no idea in this sort() case. It cannot be trunked, cannot run in batch. I have to sort the whole data range at one time. 

Anyone could help? 

```python
# Your traceback here
Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/aeosa/aem/aemsend.py", line 74, in send
    replyevent = self._sendproc(self.AEM_event, flags, timeout)
  File "/usr/local/lib/python3.9/site-packages/aeosa/aem/aemsend.py", line 23, in sendappleevent
    return evt.send(flags, timeout)
aem.ae.MacOSError: -1712

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.9/site-packages/aeosa/appscript/reference.py", line 479, in __call__
    return self.AS_appdata.target().event(self._code, params, atts, codecs=self.AS_appdata).send(timeout, sendflags)
  File "/usr/local/lib/python3.9/site-packages/aeosa/aem/aemsend.py", line 77, in send
    raise EventError(err.args[0]) from err
aem.aemsend.EventError: Command failed: Apple event timed out. (-1712)

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/alexela/SynologyDrive/Lonch_Home/05. programming/update-contract.py", line 304, in <module>
    refresh_formulas(wb)
  File "/Users/alexela/SynologyDrive/Lonch_Home/05. programming/update-contract.py", line 158, in refresh_formulas
    sht.range((data_row_start, data_col_start), (data_row_end, data_col_end)) \
  File "/usr/local/lib/python3.9/site-packages/aeosa/appscript/reference.py", line 515, in __call__
    raise CommandError(self, (args, kargs), e, self.AS_appdata) from e
appscript.reference.CommandError: Command failed:
		OSERROR: -1712
		MESSAGE: Apple event timed out.
		COMMAND: app(pid=8587).workbooks[2].worksheets['Data'].cells['$B$5:$IN$17153'].sort(key1=app(pid=8587).workbooks[2].worksheets['Data'].cells['$HV$5:$HV$17153'], order1=1, orientation=1)
```

#### Include a minimal code sample to reproduce the issue (and attach a sample workbook if required!)

```python
# Your code here
sort_range = sht.range((data_row_start, sort_col), (data_row_end, sort_col))
shut.range ((data_row_start, data_col_start), (data_row_end, data_col_end)).api.sort(key1=sort_range.api, order1=1, orientation=1)
```

**Comment by tsusoft** [comment]
Finally confirmed it is not an issue of xlwings. 

Excel does running with xlwings command, when python raise an exception. I checked the source code (reference.py) where it raised timeout exception, the __call__ method of class Command read ‘timeout’ from kargs, default is 60 sec. So I specified ‘timeout’ to 180s (Excel usually completes sorting in 2 mins), like rng.api.sort(key1=sort_range.api, order1=1, orientation=1, timeout=180). It works. I specified timeout to 5s, it raised exception immediately after 5 sec. That means, ‘timeout’ argument in function call took effect.

This perfectly solved my problem.  Well, I should dig more before submit an issue here. Apologize and wish this could help others.
